File: Estruturas e Bibliotecas/2633 - Churras no Yuri.cpp
#include <bits/stdc++.h>

using namespace std;

#define ft first
#define sd second
#define sz(x) (int)x.size()
#define all(x) x.begin(), x.end()
#define boost ios_base::sync_with_stdio(false), cin.tie(NULL)
#define inp freopen("code.in", "r+", stdin)
#define out freopen("code.out", "w+", stdout)
#define debug(x) cout << "DEBUG: " << x << endl

typedef vector<int> vi;
typedef pair<int, int> ii;
typedef long long int ll;

struct churras {
//Properties/Variables
    string meat;
    int validity;
    
//Methods
    string ToString() {
        return meat;
    }
};

int main()
{
    //inp;
    int n;
    while (scanf("%d", &n) != EOF) {
        const int sz = n;
        churras C[sz];  
        for (int i = 0; i < n; i++) {
            cin >> C[i].meat;
            scanf("%d", &C[i].validity);
        }
        sort(C, C + sz, [](churras a, churras b) {
            return a.validity < b.validity;
        });
        for (int i = 0; i < sz; i++)
            printf("%s%c", C[i].ToString().c_str(), ((i == n - 1) ? '\n' : ' '));
    }
}

File: Estruturas e Bibliotecas/2633 - Churras no Yuri.cs
using System;
using System.Collections.Generic;
using System.Collections;
using System.Text;
using System.Numerics;
using System.Linq;
using System.IO;

namespace ProjectsCSharp
{
    class Program 
    {
        public struct churras {
        //Properties/Variables
            public string meat { get; set; }
            public int validity { get; set; }
        
        //Methods
            public override string ToString() {
                return meat;
            }
        };

        static int mycomparator(churras a, churras b)
        {
            if (a.validity < b.validity)
                return -1;
            else
                return (a.validity == b.validity ? 0 : 1); 
        }

        static void Main(string[] args)
        {
            //Console.SetIn(new StreamReader(@"/home/hugo/projects/code.in"));
            string[] input;
            string values = Console.ReadLine();
            int n;
            churras[] C = new churras[10];
            while (values != null) {
                input = values.Split(' ');
                n = Int32.Parse(input[0]);
                Array.Resize(ref C, n);
                for (int i = 0; i < n; i++) {
                    input = Console.ReadLine().Split(' ');
                    C[i].meat = input[0];
                    C[i].validity = Int32.Parse(input[1]);
                }
                Array.Sort(C, mycomparator);
                for (int i = 0; i < n; i++)
                    Console.Write(C[i].ToString() + ((i == n - 1) ? "\n" : " "));
                values = Console.ReadLine();
            }
        }
    }
}
